libs/platform/ebpf_epoch.c

```diff
@@ -40,24 +40,37 @@
 // Delay after the _ebpf_flush_timer is set before it runs.
 #define EBPF_EPOCH_FLUSH_DELAY_IN_MICROSECONDS 1000
 
+// Time before logging that a thread entry is stale
+#define EBPF_EPOCH_STALE_THREAD_TIME_IN_NANO_SECONDS 10000000000 // 10 second
+
 typedef struct _ebpf_epoch_state
 {
     int64_t epoch;
-    bool active;
+    bool active : 1;
+    bool timer_armed : 1;
+    bool stale : 1;
+    bool shutdown : 1;
 } ebpf_epoch_state_t;
 
 // Table to track per CPU state.
 // This table must fit into a multiple of EBPF_CACHE_LINE_SIZE.
 typedef struct _ebpf_epoch_cpu_entry
 {
     ebpf_lock_t lock;
-    _Requires_lock_held_(lock) ebpf_epoch_state_t cpu_epoch_state;
+    _Requires_lock_held_(lock) ebpf_epoch_state_t epoch_state;
     _Requires_lock_held_(lock) ebpf_list_entry_t free_list;
     _Requires_lock_held_(lock) ebpf_hash_table_t* thread_table;
-    int32_t flags;
     ebpf_non_preemptible_work_item_t* stale_worker;
+    uint32_t padding;
 } ebpf_epoch_cpu_entry_t;
 
+typedef struct _ebpf_epoch_thread_entry
+{
+    ebpf_epoch_state_t epoch_state;
+    uintptr_t old_thread_affinity_mask;
+    uint64_t last_used;
```

Comment:
It was hard for me to understand what this field meant.   Maybe
```suggestion
    uint64_t last_used_time;
```